backend/src/apiserver/server/pipeline_server.go

```diff
@@ -118,3 +118,82 @@ func ValidateCreatePipelineRequest(request *api.CreatePipelineRequest) error {
 func NewPipelineServer(resourceManager *resource.ResourceManager) *PipelineServer {
 	return &PipelineServer{resourceManager: resourceManager, httpClient: http.DefaultClient}
 }
+
+func (s *PipelineServer) CreatePipelineVersion(ctx context.Context, request *api.CreatePipelineVersionRequest) (*api.PipelineVersion, error) {
+	// Read pipeline file.
+	if request.Version == nil || request.Version.PackageUrl == nil ||
+		len(request.Version.PackageUrl.PipelineUrl) == 0 {
+		return nil, util.NewInvalidInputError(
+			"Pipeline URL is empty. Please specify a valid URL.")
+	}
+	if _, err := url.ParseRequestURI(request.Version.PackageUrl.PipelineUrl); err != nil {
+		return nil, util.NewInvalidInputError(
+			"Invalid Pipeline URL %v. Please specify a valid URL",
+			request.Version.PackageUrl.PipelineUrl)
+	}
+	pipelineUrl := request.Version.PackageUrl.PipelineUrl
+	resp, err := s.httpClient.Get(pipelineUrl)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return nil, util.NewInternalServerError(err,
+			`Failed to download the pipeline from %v.
+			Please double check the URL is valid and can be accessed by the
```

Comment:
seems line breaks too eagerly (<80 char) and breaks the sentence
see best practice
https://github.com/golang/go/wiki/CodeReviewComments#line-length

please change other places in the PR too

Reply:
i'll try but can't guarantee that "a break feeling natural or unnatural", which imho quite objective. I mostly followed hard character limit per line unless completely impossible to break. I feel what might be better is to use some go fmt tool. I assume we mostly use gofmt? What I can do is: if gofmt doesn't break lines I'll never add breaks manually.

Reply:
here it's breaking in the middle of a sentence.

Reply:
ack